hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1641,7 +1641,7 @@ void optimizeARM(Vunit& unit, const Abi& abi, bool regalloc) {
   simplify(unit);
 
   lowerForARM(unit);
-
+  reuseImmq(unit);
```

Comment:
@cmuellner - I pretty much agree with all these points.  The initial intent was to have something that could start cleaning up the extra instructions around the addressing.
Ideally, there'd be a vixl callout to address \#2. above.   I suggest getting something in that's
correct and then improving forever.

Since I have to change the code again anyway, I can move that one call down.